core/key.go

```diff
@@ -304,6 +314,52 @@ func NewCollectionSchemaVersionKeyFromString(key string) (CollectionSchemaVersio
 	}, nil
 }
 
+// NewCollectionPolicyKey creates a new CollectionPolicyKey from a collectionID.
+func NewCollectionPolicyKey(
+	colID uint32,
+) CollectionPolicyKey {
+	return CollectionPolicyKey{
+		CollectionID: colID,
+	}
+}
+
+// NewCollectionPolicyKeyFromString creates a new CollectionPolicyKey from a string.
+// It expects the input string in the following format:
+//
+// /collection/policy/[CollectionID]
+//
+// Where [CollectionID] must not be omitted.
+func NewCollectionPolicyKeyFromString(key string) (CollectionPolicyKey, error) {
+	keyElements := strings.Split(key, "/")
+	if len(keyElements) != 4 || keyElements[1] != COLLECTION || keyElements[2] != "policy" {
+		return CollectionPolicyKey{}, ErrInvalidKey
+	}
+
+	colID, err := strconv.Atoi(keyElements[3])
+	if err != nil {
+		return CollectionPolicyKey{}, err
+	}
+
+	return CollectionPolicyKey{
+		CollectionID: uint32(colID),
+	}, nil
+}
+
+// ToString returns the string representation of the key
+// It is in the following format:
+// /collection/policy/[CollectionID]
+func (k CollectionPolicyKey) ToString() string {
+	return fmt.Sprintf("%s/%s", COLLECTION_POLICY, fmt.Sprint(k.CollectionID))
```

Comment:
todo:  Please don't do it like this (I know we have a couple of others like this in this file) :)

```
return COLLECTION_POLICY + "/" + strconv.Itoa(int(k.CollectionID)))
```
Is my preference atm, but I guess you could do:
```
return fmt.Sprintf("%s/%v", COLLECTION_POLICY, k.CollectionID)
```

Check my other comments first though, this key type should be removed IMO.

Reply:
- [ ] Do the `return COLLECTION_POLICY + "/" + strconv.Itoa(int(k.CollectionID)))`

- [ ] Check if key should be removed.

Reply:
- [x] Not relevant anymore the key was removed completely.